src/Compilers/VisualBasic/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.vb

```diff
@@ -0,0 +1,303 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.CodeGen
+Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Roslyn.Utilities
+Imports System.Collections.Immutable
+Imports System.Diagnostics
+
+Namespace Microsoft.CodeAnalysis.VisualBasic
+
+    ''' <summary>
+    ''' This type provides means for instrumenting compiled methods for dynamic analysis.
+    ''' It can be combined with other <see cref= "Instrumenter" /> s.
+    ''' </summary>
+    Friend NotInheritable Class DynamicAnalysisInjector
+        Inherits CompoundInstrumenter
+
+        Private ReadOnly _method As MethodSymbol
+        Private ReadOnly _methodBody As BoundStatement
+        Private ReadOnly _createPayload As MethodSymbol
+        Private ReadOnly _spansBuilder As ArrayBuilder(Of SourceSpan)
+        Private _dynamicAnalysisSpans As ImmutableArray(Of SourceSpan) = ImmutableArray(Of SourceSpan).Empty
+        Private ReadOnly _methodEntryInstrumentation As BoundStatement
+        Private ReadOnly _payloadType As ArrayTypeSymbol
+        Private ReadOnly _methodPayload As LocalSymbol
+        Private ReadOnly _diagnostics As DiagnosticBag
+        Private ReadOnly _debugDocumentProvider As DebugDocumentProvider
+        Private ReadOnly _methodHasExplicitBlock As Boolean
```

Comment:
After a few days of pondering I've convinced myself that instrumenting VB does not require this information.